backend/ravegenieApp/controllers/TransactionController.py

```diff
@@ -28,9 +28,9 @@ def deposit(self, request, service_name):
 			comment=service_name,
 			dollar_amount=deposit_form.cleaned_data['amount']
 		)
-		try:  
+		try:
 			gateway_url = payment_service.get_payment_provider().init_payment(
-				amount=transaction.naira_amount,
+				amount=str(transaction.naira_amount),
```

Comment:
Coercing the amount to string when it's called leaks the abstraction of the init_payment. That modification should have been inside the init_payment class so the caller doesn't need to know or accidentally forget